sdk/identity/identity/README.md

```diff
@@ -59,6 +59,23 @@ For systems without a default web browser, the `az login` command will use the d
 
 To authenticate Azure SDKs within web browsers, we currently offer the `InteractiveBrowserCredential`, which can be set to use redirection or popups to complete the authentication flow. It is necessary to [create an Azure App Registration](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration) in the portal for your web application first.
 
+In the browser, two authentication flows are provided:
+
+- OAuth 2.0's [Auth Code Flow][AuthCodeFlow], which is the preferred flow, uses [Proof Key for Code Exchange (PKCE)](https://tools.ietf.org/html/rfc7636), uses [MSAL v2.x](https://github.com/AzureAD/microsoft-authentication-library-for-js), and is enabled by default.
+- OAuth 2.0's [Implicit Grant Flow][ImplicitGrantFlow], which is an older authentication method that uses third party cookies (which are set to be removed from browsers at some point).
+
+While [Auth Code Flow][AuthCodeFlow] is the recommended authentication flow, [Implicit Grant Flow][ImplicitGrantFlow] is still available by passing the `flow` property with the value `implicit-grant` through to the constructor of the `InteractiveBrowserCredential`.
+
+The `InteractiveBrowserCredential` allows users to pass an optional `redirectUri` property that is assumed to be the website's URL if not provided. This redirect URI should be configured in your Azure Active Directory App Registration. Follow the instructions for [creating your single-page application](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration#redirect-uri-msaljs-20-with-auth-code-flow) to correctly mark your redirect URI as enabled for CORS.
+
+If you attempt to use the authorization code flow and see this error:
+
+```
+access to XMLHttpRequest at 'https://login.microsoftonline.com/common/v2.0/oauth2/token' from origin 'yourApp.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
+```
+
+Then you need to visit your app registration and update the redirect URI for your app to type `spa` (for "single page application").
+
```

Comment:
I feel that this might be a bit too much detail for here, especially considering this comes before even `Key Concepts`. This section is intended to contain information about how to authenticate in your development environment (Given this I'm not sure the existing information is correct, because the `InteractiveBrowserCredential` for node implementation should use a default client id, but that's a separate issue). I think it would be useful to encapsulate this information into it's own, doc with samples and we can then link to it from the readme as well as the changelog where you're introducing the breaking change so users can quickly get information on how they might be impacted.

Reply:
What about this?

---

### Authenticate the client in browsers

To authenticate Azure SDKs within web browsers, we currently offer the `InteractiveBrowserCredential`, which can be set to use redirection or popups to complete the authentication flow.

The `InteractiveBrowserCredential` uses [Auth Code Flow][AuthCodeFlow], which uses [Proof Key for Code Exchange (PKCE)](https://tools.ietf.org/html/rfc7636), uses [MSAL v2.x](https://github.com/AzureAD/microsoft-authentication-library-for-js), and is enabled by default. It also allows switching back to the [Implicit Grant Flow][ImplicitGrantFlow] by passing the `flow` property with the value `implicit-grant` through to the constructor of the `InteractiveBrowserCredential`.

Follow the instructions for [creating your single-page application](https://docs.microsoft.com/azure/active-directory/develop/scenario-spa-app-registration#redirect-uri-msaljs-20-with-auth-code-flow) to correctly mark your redirect URI as enabled for CORS.

If you attempt to use the authorization code flow and see this error:

```
access to XMLHttpRequest at 'https://login.microsoftonline.com/common/v2.0/oauth2/token' from origin 'yourApp.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

Then you need to visit your app registration and update the redirect URI for your app to type `spa` (for "single page application").

Reply:
I'll move this to a separate file!

Reply:
I moved it to a separate file.